pkg/rpc/server.go

```diff
@@ -0,0 +1,119 @@
+package smrpc
+
+import (
+	fmt "fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	empty "github.com/golang/protobuf/ptypes/empty"
+	"golang.org/x/net/context"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+	mount "k8s.io/mount-utils"
+
+	iscsiutil "github.com/longhorn/go-iscsi-helper/util"
+
+	"github.com/longhorn/longhorn-share-manager/pkg/server"
+	"github.com/longhorn/longhorn-share-manager/pkg/types"
+	"github.com/longhorn/longhorn-share-manager/pkg/volume"
+)
+
+type ShareManagerServer struct {
+	manager *server.ShareManager
+}
+
+func NewShareManagerServer(manager *server.ShareManager) *ShareManagerServer {
+	return &ShareManagerServer{
+		manager: manager,
+	}
+}
+
+func (s *ShareManagerServer) FilesystemTrim(ctx context.Context, req *FilesystemTrimRequest) (resp *empty.Empty, err error) {
+	volumeName := s.manager.Volume.Name
+
+	defer func() {
+		if err != nil {
+			logrus.WithError(err).Errorf("failed to trim mounted filesystem on volume %v", volumeName)
+		}
+	}()
+
+	devicePath := types.GetVolumeDevicePath(volumeName, req.IsEncryptedDevice)
+	if !volume.CheckDeviceValid(devicePath) {
+		return &empty.Empty{}, fmt.Errorf("volume %v is not valid", volumeName)
+	}
+
+	mountPath := types.GetMountPath(volumeName)
```

Comment:
NIT: Do we need to check if the device path matches the mount path?

Reply:
Do you mean check magic number, right?